src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.Figure;
+import core.basesyntax.enums.Color;
+import core.basesyntax.enums.FigureType;
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_SIZE = 9;
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private static final FigureType[] FIGURE_ARRAY = FigureType.values();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    private FigureType figuresType;
+
+    private int getRandomSize() {
+        return random.nextInt(MAX_SIZE) + 1;
+    }
+
+    public Figure getRandomFigure() {
+        int figureIndex = random.nextInt(FIGURE_ARRAY.length);
+        figuresType = FIGURE_ARRAY[figureIndex];
+        Color color = colorSupplier.getRandomColor();
+        String colorName = color.name().toLowerCase();
```

Comment:
return `color.name().toLowerCase()` from `getRandomColor()` method
```suggestion
        String color = colorSupplier.getRandomColor();
```